simulatingrisk/hawkdove/model.py

```diff
@@ -153,19 +173,25 @@ class HawkDoveModel(mesa.Model):
     min_window = 15
     #: class to use when initializing agents
     agent_class = HawkDoveAgent
+    #: supported neighborhood sizes
+    neighborhood_sizes = {4, 8, 24}
 
     def __init__(
         self,
         grid_size,
-        include_diagonals=True,
+        play_neighborhood=8,
         hawk_odds=0.5,
     ):
         super().__init__()
         # assume a fully-populated square grid
         self.num_agents = grid_size * grid_size
-        # mesa get_neighbors supports moore neighborhood (include diagonals)
-        # and von neumann (exclude diagonals)
-        self.include_diagonals = include_diagonals
+        check_neighborhood_size(play_neighborhood)
+        self.play_neighborhood = play_neighborhood
+        # since we define risk attitudes from 0-8 or 0-4,
+        # the neighborhood for choosing what to play cannot be bigger than 8;
+        # use the smaller of the play neighborhood or 8
+        self.choose_neighborhood = min(8, play_neighborhood)
```

Comment:
this attribute name sounds like a function, but it's clear enough here

Reply:
oh wow, it totally does - I didn't think of the fact that it sounds like a verb. I think it's probably ok in context, but I'll keep in mind if Lara or I can come up with a better name. (She had much longer descriptions in her write-up that I was trying to condense)

Reply:
On this, I think the neighborhood that determines what the agent should play should be 8 (or 4 in the "don't include diagonals" case), even if the play neighborhood is smaller. The intuitive idea behind the "choose neighborhood" is that the agent looks at it to determine the probability of a random agent playing HAWK, and uses that in her future interactions. It doesn't need to overlap with the agents she's actually playing, and there's an interesting case in which this probability is determined by a *larger* sample than the one she interacts with, e.g., she uses 8 neighbors to determine the probability but only interacts with 4. (You can imagine, for example, making predictions about what someone playing a game will do, even if you've never played with that particular person, by thinking about what's happened when you've played that game in the past with different people.) Happy to explain more if this doesn't make sense.

This is also where the assumption that Kevin called "myopia" entered in. A different way for an agent to determine the probability that some randomly-selected agent plays HAWK might be to look at what her 8 neighbors have done on every round so far, or on the last 10 rounds, or whatever. (Another potential thing to try, though it wasn't a priority--I'll add it to our list.)

Maybe it would be clearer if we called this "evidence neighborhood," since it's the neighborhood that serves as evidence for the agent to set her probabilities.

Reply:
@LaraBuchak thanks for responding. I wondered about calling it "observed neighborhood" (relates to seen / myopia) but wondered if that was ambiguous with the neighbors used for deciding how to adjust risk attitudes, which is a different kind of observation.

I have a sense of the potential interesting cases here where the people you observe are not necessarily the same set as the people you play against, I think I just misunderstood what you wanted in your write-up. Do you want me to make the evidence neighborhood size a user-configurable parameter? And is there a reason not to allow it to go up to 24? It shouldn't be too hard to adjust the choice logic for different numbers of neighbors and risk attitude scale (and we'll need to do that anyway if they don't always match).